mirrulations-client/src/mirrclient/disk_saver.py

```diff
@@ -70,3 +70,42 @@ def is_duplicate(self, existing, new):
     def check_for_duplicates(self, path, data, i):
         if self.is_duplicate(self.open_json_file(path), data) is False:
             self.save_duplicate_json(path, data, i)
+
+    def get_extracted_text_attachments_dir(self, path):
+        # remove file name from path
+        path = path.rsplit("/", 1)[0]
+        return path.replace('comments_extracted_text/pdfminer',
+                            'comments_attachments/').replace('text', 'binary')
+
+    def generate_metadata_for_extraction(self, path):
+        # comments_extracted_text structure
+        # data/data/EPA/EPA-HQ-OA-2003-0001/text-EPA-HQ-OA-2003-0001/comments_extracted_text/pdfminer/
+        # Attachments directory for given comment
+        # data/data/EPA/EPA-HQ-OA-2003-0001/binary-EPA-HQ-OA-2003-0001/comments_attachments
+
+        docket_id = path.rsplit("/comments_extracted_text", 1)[0]\
+            .split("text-")[-1]
+
+        attachments_dir = self.get_extracted_text_attachments_dir(path)
+        extracted_dir = path.rsplit("/", 1)[0]
+        pdf_files = [f for f in os.listdir(attachments_dir) if ".pdf" in f]
+        extracted_txt_files = [file for file in os.listdir(extracted_dir)
+                               if ".txt" in file]
+        meta = {
```

Comment:
The meta data file should really have information per attachment.

ex.
{
"extraction_status": {
    "name_of_attachment_1.pdf": "success",
    "name_of_attachment_2.pdf": "failure"
    "name_of_attachment_3.doc": "not attempted"
    }
}

Or something similar at the very least. In its current state, We are only giving total numbers.